src/librustc_mir/borrow_check/diagnostics/move_errors.rs

```diff
@@ -397,6 +398,16 @@ impl<'a, 'tcx> MirBorrowckCtxt<'a, 'tcx> {
                     format!("{}.as_ref()", snippet),
                     Applicability::MaybeIncorrect,
                 );
+            } else if span.is_desugaring(DesugaringKind::ForLoop)
+                && move_ty.starts_with("std::vec::Vec")
```

Comment:
Why not use `diagnostic_item`s?

Reply:
Just keeping in line with the code above this.

Reply:
I think we should avoid introducing more tech debt if we can and instead elect to clean up code before adding enhancements. If there isn't a good reason to avoid diagnostic items (which `Vec` already is) I think they should be used. cc @oli-obk

Reply:
If it isn't too difficult to do, I think it would be nice to see these `move_ty.starts_with` checks replaced with diagnostic items.

Reply:
Done. Looking at this it seems this code predates `diagnostic_item`.